diff --git a/lib/net_buf/buf_simple.c b/lib/net_buf/buf_simple.c
index 9fb0ced6dab..63309fbe9f1 100644
--- a/lib/net_buf/buf_simple.c
+++ b/lib/net_buf/buf_simple.c
@@ -15,6 +15,7 @@ LOG_MODULE_REGISTER(net_buf_simple, CONFIG_NET_BUF_LOG_LEVEL);
 #include <zephyr/sys/byteorder.h>
 
 #include <zephyr/net_buf.h>
+#include "fwconfig_cmn.h"
 
 #if defined(CONFIG_NET_BUF_SIMPLE_LOG)
 #define NET_BUF_SIMPLE_DBG(fmt, ...) LOG_DBG("(%p) " fmt, k_current_get(), \
@@ -70,7 +71,12 @@ void *net_buf_simple_add_mem(struct net_buf_simple *buf, const void *mem,
 {
 	NET_BUF_SIMPLE_DBG("buf %p len %zu", buf, len);
 
+#ifdef MEM_CPY_VIA_DXE
+	extern void *nt_dpm_memcpy(void *dst, const void *src, uint32_t length);
+	return nt_dpm_memcpy(net_buf_simple_add(buf, len), mem, len);
+#else
 	return memcpy(net_buf_simple_add(buf, len), mem, len);
+#endif
 }
 
 uint8_t *net_buf_simple_add_u8(struct net_buf_simple *buf, uint8_t val)
