diff --git a/lib/net_buf/buf.c b/lib/net_buf/buf.c
index 7b89e89fd7b..5f341ac8ad6 100644
--- a/lib/net_buf/buf.c
+++ b/lib/net_buf/buf.c
@@ -19,6 +19,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <zephyr/sys/byteorder.h>
 
 #include <zephyr/net_buf.h>
+#include "fwconfig_cmn.h"
 
 #if defined(CONFIG_NET_BUF_LOG)
 #define NET_BUF_DBG(fmt, ...) LOG_DBG("(%p) " fmt, k_current_get(), \
@@ -656,7 +657,12 @@ size_t net_buf_linearize(void *dst, size_t dst_len, const struct net_buf *src,
 	copied = 0;
 	while (frag && len > 0) {
 		to_copy = MIN(len, frag->len - offset);
+#ifdef MEM_CPY_VIA_DXE
+		extern void *nt_dpm_memcpy(void *dst, const void *src, uint32_t length);
+		nt_dpm_memcpy((uint8_t *)dst + copied, frag->data + offset, to_copy);
+#else
 		memcpy((uint8_t *)dst + copied, frag->data + offset, to_copy);
+#endif
 
 		copied += to_copy;
 
