diff --git a/subsys/net/ip/tcp.c b/subsys/net/ip/tcp.c
index a057c333e21..b2e137beeb3 100644
--- a/subsys/net/ip/tcp.c
+++ b/subsys/net/ip/tcp.c
@@ -4822,3 +4822,26 @@ void net_tcp_init(void)
 	k_thread_name_set(&tcp_work_q.thread, "tcp_work");
 	NET_DBG("Workq started. Thread ID: %p", &tcp_work_q.thread);
 }
+
+int net_tcp_stop(void)
+{
+	int ret;
+
+    ret = k_work_queue_drain(&tcp_work_q, true);
+	if(ret < 0){
+		return ret;
+	}
+
+	ret = k_work_queue_stop(&tcp_work_q, K_FOREVER);
+	return ret;
+}
+
+void net_tcp_start(void)
+{
+#if defined(CONFIG_NET_TC_THREAD_COOPERATIVE)
+#define THREAD_PRIORITY K_PRIO_COOP(CONFIG_NET_TCP_WORKER_PRIO)
+#else
+#define THREAD_PRIORITY K_PRIO_PREEMPT(CONFIG_NET_TCP_WORKER_PRIO)
+#endif
+    k_work_queue_start(&tcp_work_q, work_q_stack, K_THREAD_STACK_SIZEOF(work_q_stack), THREAD_PRIORITY, NULL);
+}
\ No newline at end of file
