diff --git a/tests/drivers/rtc/rtc_api/src/main.c b/tests/drivers/rtc/rtc_api/src/main.c
index 1c5ce94429e..271f33e6863 100644
--- a/tests/drivers/rtc/rtc_api/src/main.c
+++ b/tests/drivers/rtc/rtc_api/src/main.c
@@ -6,4 +6,12 @@

 #include <zephyr/ztest.h>

+ZTEST_SUITE(rtc_api_first, NULL, NULL, NULL, NULL, NULL);
 ZTEST_SUITE(rtc_api, NULL, NULL, NULL, NULL, NULL);
+
+void test_main(void)
+{
+	/* Force execution order */
+	ztest_run_test_suite(rtc_api_first, false, 1, 1, NULL);
+	ztest_run_test_suite(rtc_api, false, 1, 1, NULL);
+}
